src/tonic/mod.rs

```diff
@@ -69,9 +74,25 @@ impl QdrantInternal for QdrantInternalService {
         _request: Request<HttpPortRequest>,
     ) -> Result<Response<HttpPortResponse>, Status> {
         Ok(Response::new(HttpPortResponse {
-            port: self.http_port as i32,
+            port: self.settings.service.http_port as i32,
         }))
     }
+
+    async fn wait_on_consensus_commit(
+        &self,
+        request: Request<WaitOnConsensusCommitRequest>,
+    ) -> Result<Response<WaitOnConsensusCommitResponse>, Status> {
+        let request = request.into_inner();
+        let commit = request.commit as u64;
+        let term = request.term as u64;
+        let timeout = Duration::from_secs(request.timeout as u64);
```

Comment:
I'd prefer:
- either rename `request.timeout` to `request.timeout_secs`
- or, better, `request.timeout_ms` (in milliseconds)